agents/hooks/addCurrentAgent.js

```diff
@@ -0,0 +1,19 @@
+const merge = require('ramda/src/merge')
+const isNil = require('ramda/src/isNil')
+
+function addCurrentAgent(hook) {
+  if (isNil(hook.params.credential.agentId)) return hook
+
+  const agents = hook.app.service('agents')
+
+  return agents.get(hook.params.credential.agentId)
+    .then((agent) => {
+      return merge(hook, {
```

Comment:
i'm not sure if this matters, but might be best to do `hook.params =`, only because that's what the feathers docs do (which would only make a difference if the "hook object" were an instance of a special Hook class or some such).